### PR TITLE
Websocket setup

### PR DESCRIPTION
**Pull request by TofuSupreme**
Added web socket to instance. Check the Chrome console to see if it the connection works properly! 

At the moment, a host can create the lobby and will generate a connection with a new instance number (which is displayed in the Chrome console). Since the [Join game] feature is not created just yet, I did not test to see if someone else can join the same game with the same instance. 
image

### CODE REVIEW

File: app/javascript/channels/instance_channel.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import consumer from "./consumer";
+
+const initInstanceChannel = () => {
+  console.log("Connection complete");
+  const instanceContainer = document.getElementById("instance")
+  if (instanceContainer) {
+    const id = instanceContainer.dataset.instanceId;
+    console.log("Instance id: ", id);
+
+    consumer.subscriptions.create({ channel: "InstanceChannel" }, {
```
Comment (carlnoval) [inline app/javascript/channels/instance_channel.js:10]: We seem to be missing to link the channel id here from line 7. [Video reference from lecture](https://kitt.lewagon.com/camps/718/lectures/06-Projects%2F01-Pundit#video) at time: 40:40.

Not having the id is ok if we always expect there to always only be one channel in the app. However multiple channels/dbi may be created at a time.

File: app/javascript/channels/instance_channel.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import consumer from "./consumer";
+
+const initInstanceChannel = () => {
+  console.log("Connection complete");
+  const instanceContainer = document.getElementById("instance")
+  if (instanceContainer) {
+    const id = instanceContainer.dataset.instanceId;
+    console.log("Instance id: ", id);
```
Comment (carlnoval) [inline app/javascript/channels/instance_channel.js:8]: In order to know if the user has successfully subscribed to the channel, I think this line should be inside `connected() {` from line 11.

**Comment by carlnoval** [comment]
As for the conflict, I think placing this inside the `<div id="instance" data-instance-id="<%=@instance.id%>">...</div>` should do the trick. Would recommend to resolve this conflict after the other review comments though.
```ruby
<h2>Players:</h2>
<ul>
  <% @players.each do |player| %>
    <% user = User.find(player.user_id)%>
    <li><%= user.nickname%></li>
  <% end %>
</ul>

<h2>Pin number:</h2>
<%= @instance.pin %>
```

File: app/javascript/channels/instance_channel.js (hunk 1 above)
Comment (TofuSupreme) [reply app/javascript/channels/instance_channel.js:10]: Ah! You're right! I forgot to add that back in. Thanks for the catch.

**Comment by TofuSupreme** [comment]
> As for the conflict, I think placing the block of code below inside of `<div id="instance" data-instance-id="<%=@instance.id%>">...</div>` should do the trick. Would recommend to resolve this conflict after the other review comments though.
> 
> ```ruby
> <h2>Players:</h2>
> <ul>
>   <% @players.each do |player| %>
>     <% user = User.find(player.user_id)%>
>     <li><%= user.nickname%></li>
>   <% end %>
> </ul>
> 
> <h2>Pin number:</h2>
> <%= @instance.pin %>
> ```

Does this work?? On my end, I've imp

**Comment by spioupiou** [comment]
FYI - the "join lobby" code works only if you're logged in for the moment! So maybe that's why you had to comment it out while testing

**Comment by carlnoval** [comment]
I also think that this works now. Will merge.

**Pull request merged**